tests/tracer/test_logging.py

```diff
@@ -0,0 +1,356 @@
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+import ddtrace
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err != b""
+    assert out == b""
+
+
+def test_unrelated_logger_loaded_last_in_debug(run_python_code_in_subprocess):
+    """
+    When the tracer is imported before logging has been configured in debug mode,
+        the ddtrace logger does not override any custom logs settings.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    code = """
+import ddtrace
+import logging
+
+custom_logger = logging.getLogger('custom')
+custom_logger.setLevel(logging.WARN)
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err != b""
+    assert out == b""
+
+
+def test_child_logger_inherits_settings(run_python_code_in_subprocess, tmpdir):
+    """
+    Child loggers under the ddtrace name inherit settings as expected.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.handlers.RotatingFileHandler)
+assert ddtrace_logger.handlers[0].maxBytes == 15 << 20
+assert ddtrace_logger.handlers[0].backupCount == 1
+
+child_logger = logging.getLogger('ddtrace.child')
+assert child_logger.parent.name == 'ddtrace'
+assert child_logger.getEffectiveLevel() == logging.DEBUG
+assert child_logger.handlers == []
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+        assert re.search(LOG_PATTERN, content) is not None
+
+
+def test_unrelated_logger_in_debug_with_ddtrace_run(ddtrace_run_python_code_in_subprocess, tmpdir):
+    """
+    When using ddtrace-run with a custom logger,
+        the ddtrace logger does not override any custom logs settings.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    env["DD_TRACE_FILE_SIZE_BYTES"] = "200000"
+    code = """
+import logging
+custom_logger = logging.getLogger('custom')
+custom_logger.setLevel(logging.WARN)
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
```

Comment:
since this is the same assertion code above, you can probably just collapse them into the same test case, e.g.:

```python
def test_unrelated_logger_loaded_last_in_debug(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess):
    env = os.environ.copy()
    env["DD_TRACE_DEBUG"] = "true"
    code = """
import ddtrace
import logging
custom_logger = logging.getLogger('custom')
custom_logger.setLevel(logging.WARN)
assert custom_logger.parent.name == 'root'
assert custom_logger.level == logging.WARN
"""

    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
    assert status == 0, err
    assert err != b""
    assert out == b""

    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
    assert status == 0, err
    assert err != b""
    assert out == b""
```

paired with the parametrize means you can get a ton of coverage from a single test.

it does pack a lot of different assertions (with + without ddtrace-run) into a single test case, but given the setup code and multi-line string as code, it might help maintainability in the long run.